core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileSupplier.java

```diff
@@ -105,4 +177,64 @@ static ProfileFileSupplier wrapIntoNullableSupplier(ProfileFile profileFile) {
         return Objects.nonNull(profileFile) ? () -> profileFile : null;
     }
 
+    /**
+     * Creates a {@link ProfileFileSupplier} by combining the {@link ProfileFile} objects from two {@code ProfileFileSupplier}s.
+     * Objects are passed into {@link ProfileFile.Aggregator}.
+     *
+     * @param suppliers Array of {@code ProfileFileSupplier} objects. {@code ProfileFile} objects are passed to
+     *                  {@link ProfileFile.Aggregator#addFile(ProfileFile)} in the same argument order as the supplier that
+     *                  generated it.
+     * @return Implementation of {@link ProfileFileSupplier} aggregating results from the supplier objects.
+     */
+    static ProfileFileSupplier aggregate(ProfileFileSupplier... suppliers) {
+
+        return new ProfileFileSupplier() {
+
+            final ConcurrentHashMap<ProfileFileSupplier, ProfileFile> currentValuesBySupplier = new ConcurrentHashMap<>();
+
+            @Override
+            public ProfileFile get() {
+                boolean refreshAggregate = false;
+                ProfileFile.Aggregator aggregator = ProfileFile.aggregator();
+                for (ProfileFileSupplier supplier : suppliers) {
+                    if (updateCurrentValue(supplier, aggregator::addFile)) {
+                        refreshAggregate = true;
+                    }
+                }
+
+                if (refreshAggregate) {
+                    refreshAggregate(aggregator);
+                }
+
+                return currentValuesBySupplier.get(this);
+            }
+
+            @Override
+            public void close() {
+                currentValuesBySupplier.remove(this);
+                currentValuesBySupplier.keySet().forEach(ProfileFileSupplier::close);
+            }
+
+            private boolean updateCurrentValue(ProfileFileSupplier supplier, Consumer<ProfileFile> action) {
+                ProfileFile next = supplier.get();
+                ProfileFile current = currentValuesBySupplier.put(supplier, next);
+                action.accept(next);
+
+                return !Objects.equals(next, current);
```

Comment:
These checks seem redundant. The contract of the API is that `ProfileFileSupplier#get` should return the same object if it's not updated

Reply:
Agree. Entire method has been removed.